File: 백준/Silver/15652. N과 M （4）/N과 M （4）.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
    static int N, M;
    static int[] arr;
    static StringBuilder sb = new StringBuilder();

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        br.close();

        N = Integer.parseInt(st.nextToken());
        M = Integer.parseInt(st.nextToken());

        arr = new int[M];

        solution(1, 0);

        System.out.println(sb);
    }

    static void solution(int start, int depth) {
        if (depth == M) {
            for (int i : arr) {
                sb.append(i).append(' ');
            }
            sb.append('\n');
            return;
        }

        for (int i = start; i <= N; i++) {
            arr[depth] = i;
            solution(i, depth + 1);
        }
    }
}

File: 백준/Silver/15652. N과 M （4）/N과 M （4）.py
from sys import stdin
input = stdin.readline

def solution(start: int, depth: int):
    if depth == M:
        print(' '.join(map(str, arr)))
        return

    for i in range(start, N + 1):
        arr[depth] = i
        solution(i, depth + 1)

N, M = map(int, input().split())
arr = [0] * M
solution(1, 0)